### ISSUE TITLE
An empty section in configuration raise a non explicit error

### ISSUE THREAD

**Issue by bankair** [bbatsov/rubocop#1782]
```
$ cat .rubocop.yml
AllCops:
  RunRailsCops: true

Rails/HasAndBelongsToMany:
$ touch foobar.rb
$ rubocop foobar.rb
Inspecting 1 file


0 files inspected, no offenses detected
undefined method `each_key' for nil:NilClass
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config.rb:177:in `block in validate_parameter_names'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config.rb:176:in `each'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config.rb:176:in `validate_parameter_names'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config.rb:111:in `validate'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config.rb:80:in `warn_unless_valid'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config_loader.rb:41:in `load_file'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config_loader.rb:85:in `configuration_from_file'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/config_store.rb:40:in `for'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:127:in `inspect_file'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:100:in `block in do_inspection_loop'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:94:in `loop'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:94:in `do_inspection_loop'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:72:in `process_file'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:53:in `block in inspect_files'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:51:in `each'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:51:in `inspect_files'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/runner.rb:30:in `run'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/lib/rubocop/cli.rb:24:in `run'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/bin/rubocop:13:in `block in <top (required)>'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/2.0.0/benchmark.rb:296:in `realtime'
/usr/local/var/rbenv/versions/2.0.0-p353/lib/ruby/gems/2.0.0/gems/rubocop-0.30.0/bin/rubocop:12:in `<top (required)>'
/usr/local/var/rbenv/versions/2.0.0-p353/bin/rubocop:23:in `load'
/usr/local/var/rbenv/versions/2.0.0-p353/bin/rubocop:23:in `<main>'
$
```

I'll propose a PR

**Comment by jonas054** [comment]
:+1: Good.

**Comment by bankair** [comment]
closed by PR #1784
